scripts/setup/nrfconnect/update_ncs.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import os
+import subprocess
+
+parser = argparse.ArgumentParser(description='Script helping to update nRF Connect SDK to currently recommended version.')
+parser.add_argument("-c", "--check", help="Check if your current nRF Connect SDK version is the same as recommended one.", action="store_true")
+parser.add_argument("-u", "--update", help="Update your nRF Connect SDK to currently recommended version.", action="store_true")
+args = parser.parse_args()
+
+chip_root = os.path.dirname(os.path.abspath(__file__)).replace("scripts/setup/nrfconnect", "")
+
+def get_repository_commit_sha(repository_location):
+    command = "git -C " + repository_location + " rev-parse HEAD"
+    process = subprocess.Popen(command.split(), stdout=subprocess.PIPE)
+    stdout, stderr = process.communicate()
+    if process.returncode:
+        raise subprocess.CalledProcessError(process.returncode, command, output=stdout, stderr=stderr)
+    else:
+        return subprocess.CompletedProcess(process.args, process.returncode, stdout, stderr).stdout.decode('ascii').strip()
+
+def update_ncs(repository_location, version):
+    # Checkout sdk-nrf to the desired version.
+    command = "git -C " + repository_location + " checkout " + version
+    process = subprocess.Popen(command.split(), stdout=subprocess.PIPE)
```

Comment:
This also looks like `subprocess.check_call`. Also, should we do `git fetch` before?

Reply:
Good point, added fetching repo.